### DISCUSSION TITLE
Authoring Manifest entry on how to update existing manifests

### DISCUSSION THREAD

**Discussion by ericoporto** [microsoft/winget-pkgs#99230] [Q&A] [answered]
https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md

Someone was updating and then stopped updating the Adventure Game Studio package here: https://github.com/microsoft/winget-pkgs/tree/master/manifests/a/AGSProjectTeam/AdventureGameStudio

I looked briefly about updating but the repository is huge and I have no idea how to update a package when a manifest already exists, it looks like it's just the download links, incrementing the version number and updating the sha256 hash. These things look like they could be automated, but I could not find any way in the guide on authoring_manifest on how to create these updates.

**Comment by Trenly** [comment, marked as answer]
> [`master`/AUTHORING_MANIFESTS.md](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md)
> 
> Someone was updating and then stopped updating the Adventure Game Studio package here: [`master`/manifests/a/AGSProjectTeam/AdventureGameStudio](https://github.com/microsoft/winget-pkgs/tree/master/manifests/a/AGSProjectTeam/AdventureGameStudio)
> 
> I looked briefly about updating but the repository is huge and I have no idea how to update a package when a manifest already exists, it looks like it's just the download links, incrementing the version number and updating the sha256 hash. These things look like they could be automated, but I could not find any way in the guide on authoring_manifest on how to create these updates.

There are tools like [winget-create](/microsoft/winget-create) which you can install with winget, configure your GitHub token, and then submit an update using `wingetcreate update <packageId> -i` (for interactive mode). There is also a PowerShell script in this repo specifically for creating and updating manifests - [YamlCreate.ps1](https://github.com/microsoft/winget-pkgs/blob/master/doc/tools/YamlCreate.md). With YamlCreate you can just create a fork of the repo, clone it to your local machine, and run the script - It will prompt for all the information it needs (Most likely a "Quick update" in this case).

**Reply by ericoporto** [reply]
I think winget-create is the way, as cloning this repository is something that I don't want to do.

**Reply by ericoporto** [reply]
Btw, I really don't want to do this, so if you could keep updating that package it would be nice - I am pushing on chocolatey.
